File: chase/treeInsert.ts
import TreeNode from "./treeNode";
export default function buildTreeLeetCode(inputArray: any[]): TreeNode {
    const root = new TreeNode(inputArray.shift());
    const queue = [root];
    while (queue.length > 0 && inputArray.length > 0) {
      const curNode = queue.shift();
      const leftVal = inputArray.shift() ?? null;
      const rightVal = inputArray.shift() ?? null;
      if (leftVal !== null) {
        curNode!.left = new TreeNode(leftVal);
        queue.push(curNode!.left);
      }
      if (rightVal !== null) {
        curNode!.right = new TreeNode(rightVal);
        queue.push(curNode!.right);
      }
    }
  
    return root;
}

File: easy/treeInsert.py
from typing import List
from treeNode import TreeNode

def buildTree(inputArray:List):
    root = TreeNode(inputArray.pop(0))
    q = [root]
    while len(q) > 0 and len(inputArray) > 0:
        curNode = q.pop(0)
        left = inputArray.pop(0)
        right = inputArray.pop(0)
        
        if left != None:
            curNode.left = TreeNode(left)
            q.append(curNode.left)
        if right != None:
            curNode.right = TreeNode(right)
            q.append(curNode.right)
    
    return root
